server/odc-service/src/main/java/com/oceanbase/odc/service/common/DruidMonitor.java

```diff
@@ -0,0 +1,646 @@
+            vo.setBatchSizeTotal((Long) map.get(DruidSQLStateConstants.BATCH_SIZE_TOTAL));
+            vo.setFetchRowCountMax((Long) map.get(DruidSQLStateConstants.FETCH_ROW_COUNT_MAX));
+            vo.setErrorCount((Long) map.get(DruidSQLStateConstants.ERROR_COUNT));
+            vo.setBatchSizeMax((Long) map.get(DruidSQLStateConstants.BATCH_SIZE_MAX));
+            vo.setLastErrorTime((String) map.get(DruidSQLStateConstants.LAST_ERROR_TIME));
+            vo.setReaderOpenCount((Long) map.get(DruidSQLStateConstants.READER_OPEN_COUNT));
+            vo.setEffectedRowCountMax((Long) map.get(DruidSQLStateConstants.EFFECTED_ROW_COUNT_MAX));
+            vo.setLastErrorClass((String) map.get(DruidSQLStateConstants.LAST_ERROR_CLASS));
+            vo.setInTransactionCount((Long) map.get(DruidSQLStateConstants.IN_TRANSACTION_COUNT));
+            vo.setLastErrorStackTrace((String) map.get(DruidSQLStateConstants.LAST_ERROR_STACK_TRACE));
+            vo.setResultSetHoldTime((Long) map.get(DruidSQLStateConstants.RESULT_SET_HOLD_TIME));
+            vo.setTotalTime((Long) map.get(DruidSQLStateConstants.TOTAL_TIME));
+            vo.setConcurrentMax((Long) map.get(DruidSQLStateConstants.CONCURRENT_MAX));
+            vo.setRunningCount((Long) map.get(DruidSQLStateConstants.RUNNING_COUNT));
+            vo.setFetchRowCount((Long) map.get(DruidSQLStateConstants.FETCH_ROW_COUNT));
+            vo.setMaxTimespanOccurTime((Date) map.get(DruidSQLStateConstants.MAX_TIME_SPAN_OCCUR_TIME));
+            vo.setLastSlowParameters((String) map.get(DruidSQLStateConstants.LAST_SLOW_PARAMETERS));
+            vo.setReadBytesLength((Long) map.get(DruidSQLStateConstants.READ_BYTES_LENGTH));
+            vo.setDbType((String) map.get(DruidSQLStateConstants.DB_TYPE));
+            vo.setDataSource((String) map.get(DruidSQLStateConstants.DATASOURCE));
+            vo.setSql((String) map.get(DruidSQLStateConstants.SQL));
+            vo.setLastError(map.get(DruidSQLStateConstants.LAST_ERROR));
+            vo.setMaxTimespan((Long) map.get(DruidSQLStateConstants.MAX_TIME_SPAN));
+            vo.setBlobOpenCount((Long) map.get(DruidSQLStateConstants.BLOB_OPEN_COUNT));
+            vo.setExecuteCount((Long) map.get(DruidSQLStateConstants.EXECUTE_COUNT));
+            vo.setEffectedRowCount((Long) map.get(DruidSQLStateConstants.EFFECTED_ROW_COUNT));
+            vo.setReadStringLength((Long) map.get(DruidSQLStateConstants.READ_STRING_LENGTH));
+            vo.setExecuteAndResultHoldTimeHistogram(
+                    (long[]) map.get(DruidSQLStateConstants.EXECUTE_AND_RESULT_HOLD_TIME_HISTOGRAM));
+            vo.setFile((String) map.get(DruidSQLStateConstants.FILE));
+            vo.setClobOpenCount((Long) map.get(DruidSQLStateConstants.CLOB_OPEN_COUNT));
+            vo.setLastTime((Date) map.get(DruidSQLStateConstants.LAST_TIME));
+            vo.setFetchRowCountHistogram((long[]) map.get(DruidSQLStateConstants.FETCH_ROW_COUNT_HISTOGRAM));
```

Comment:
how to guarantee that the value type of the map(Map<String, Object>)?

how should we know that the type of value(Map<String, Object>) has been changed if we upgrade the druid's version?

Reply:
Indeed, it is not possible to automatically ensure the correct correspondence of Map<String, Object> , so it needs to be manually judged whether the data has correct value by observing the logs such as method-monitor.log or sql-monitor.log. Based on the current statistical methods of Druid, there is no better approach for the time being.